src/api/wallet/WalletApiMock.ts

```diff
@@ -44,7 +48,31 @@ export class WalletApiMock implements WalletApi {
   public async getNetworkId(): Promise<number> {
     assert(this._connected, 'The wallet is not connected')
 
-    return Network.Rinkeby
+    return this._networkId
+  }
+
+  public addOnChangeWalletInfo(callback: (walletInfo: WalletInfo) => void, trigger?: boolean): void {
+    this._listeners.push(callback)
+    if (trigger) {
+      callback(this._getWalletInfo())
+    }
+  }
+
+  public removeOnChangeWalletInfo(callback: (walletInfo: WalletInfo) => void): void {
+    this._listeners = this._listeners.filter(c => c !== callback)
+  }
+
+  private _getWalletInfo(): WalletInfo {
+    return {
+      isConnected: this._connected,
+      userAddress: this._connected ? this._user : undefined,
+      networkId: this._connected ? this._networkId : undefined,
+    }
+  }
+
+  private async _notifyListeners(): Promise<void> {
+    const walletInfo: WalletInfo = this._getWalletInfo()
+    this._listeners.forEach(listener => listener(walletInfo))
```

Comment:
So your idea is to trigger listeners on every change, without specifying the event type `notifyListers('accounChanged')`?
And then listeners will have to re-query the WalletInfo state

Reply:
I was thinking on diferencienting but went for this simple approach.
We could provide the name the events, but notice that a connect event would modify the address and network.

Not a strong opinion, if you see the hook the implementation is super simple.
It may be improvable cause it can trigger more re-renders than it should, but I don't see it as a big issue.

Reply:
Not a strong opinion either :+1:

Reply:
Okay, I think as we include more APIs and components common sense will tell us what should be done :)